Makefile

```
@@ -61,6 +61,22 @@ coverage:
 	TEST_FREELIST_TYPE=array go test -v -timeout ${TESTFLAGS_TIMEOUT} \
 		-coverprofile cover-freelist-array.out -covermode atomic
 
+COMMANDS=bbolt
+
+BINARIES=$(addprefix bin/,$(COMMANDS))
+
+ALWAYS_REBUILD:
+
+bin/%: cmd/% ALWAYS_REBUILD # Build a binary from cmd/
+	@echo building $<
+	@go build -o $@ ./$<
+
+build: $(BINARIES) # Build binaries from cmd/
+
+.PHONY: clean
+clean: # Clean binaries
+	rm -f $(BINARIES)
```

Suggest to keep the makefile as simple as possible for now, and enhance it when it's really needed. If it only takes 1 minute to read/understand it, then let's do not take 2 minutes to read/understand it.

```suggestion
BOLT_CMD=bbolt

build:
	go build -o bin/${BOLT_CMD} ./cmd/bbolt

.PHONY: clean
clean: # Clean binaries
	rm -f ${BOLT_CMD}
	rm -f ./bin/${BOLT_CMD}
```

---

Updated.